Panic in Telegraf with new snmp plugin using conversion in 1.0.1

Using the new conversion option for snmp fields causes a panic in telegraf

##error

2016/10/05 12:26:30 Output [influxdb] wrote batch of 1 metrics in 30.025698ms
2016/10/05 12:26:37 Attempting connection to output: influxdb
2016/10/05 12:26:37 Successfully connected to output: influxdb
2016/10/05 12:26:37 Attempting connection to output: file
2016/10/05 12:26:37 Successfully connected to output: file
2016/10/05 12:26:37 Starting Telegraf (version 1.0.1)
2016/10/05 12:26:37 Loaded outputs: influxdb file
2016/10/05 12:26:37 Loaded inputs: snmp
2016/10/05 12:26:37 Tags enabled: host=Dell210
2016/10/05 12:26:37 Agent Config: Interval:30s, Debug:true, Quiet:false, Hostname:"Dell210", Flush Interval:30s



2016/10/05 12:27:00 Input [snmp] gathered metrics, (30s interval) in 110.824483ms
panic: unable to parse bool value '�>�F�': strconv.ParseBool: parsing "\xfa\x16>\x99F\xd1": invalid syntax


goroutine 23 [running]:
panic(0x1058660, 0xc8200fcd90)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/models.newFieldsFromBinary(0xc8204bf480, 0x13, 0x20, 0x200c82000b770)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/models/points.go:1463 +0x9f4
github.com/influxdata/influxdb/models.(*point).unmarshalBinary(0xc82000d320, 0xc820123e18)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/models/points.go:1338 +0x40
github.com/influxdata/influxdb/models.(*point).Fields(0xc82000d320, 0xc820271c60)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/models/points.go:1251 +0x3a
github.com/influxdata/influxdb/client/v2.(*Point).Fields(0xc8200fc620, 0x1f7fdc0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/influxdb/client/v2/client.go:392 +0x39
github.com/influxdata/telegraf.(*metric).Fields(0xc820024078, 0xc820271a38)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/metric.go:76 +0x24
github.com/influxdata/telegraf/agent.copyMetric(0x7f9d5afb8d98, 0xc820024078, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:293 +0x2e9
github.com/influxdata/telegraf/agent.(*Agent).flusher(0xc820024038, 0xc820276000, 0xc8202762a0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:278 +0x38f
github.com/influxdata/telegraf/agent.(*Agent).Run.func1(0xc8204c0540, 0xc820024038, 0xc820276000, 0xc8202762a0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:341 +0x7b
created by github.com/influxdata/telegraf/agent.(*Agent).Run
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:345 +0x8d5
2016/10/05 12:27:00 Attempting connection to output: influxdb
2016/10/05 12:27:00 Successfully connected to output: influxdb
2016/10/05 12:27:00 Attempting connection to output: file
2016/10/05 12:27:00 Successfully connected to output: file
2016/10/05 12:27:00 Starting Telegraf (version 1.0.1)
2016/10/05 12:27:00 Loaded outputs: influxdb file
2016/10/05 12:27:00 Loaded inputs: snmp
2016/10/05 12:27:00 Tags enabled: host=Dell210
2016/10/05 12:27:00 Agent Config: Interval:30s, Debug:true, Quiet:false, Hostname:"Dell210", Flush Interval:30s

## Telegraf config
[[inputs.snmp]]
agents = [ "172.16.1.13:161" ]
version = 2
community = "public"
interval = "30s"
timeout = "30s"

name = "pts_stats"
[[inputs.snmp.field]]
name = "hostname"
oid = ".1.3.6.1.2.1.1.5.0"
is_tag = true

[[inputs.snmp.field]]
name = "InterfaceMac"
oid = ".1.3.6.1.2.1.2.2.1.6.2"
conversion = "hwaddr"

## snmpwalk
jdavies@Dell210:~$ snmpwalk -v2c -cpublic 172.16.1.13 .1.3.6.1.2.1.2.2.1.6.2
iso.3.6.1.2.1.2.2.1.6.2 = Hex-STRING: FA 16 3E 99 46 D1

## snmpget
jdavies@Dell210:~$ snmpget -v2c -cpublic 172.16.1.13 .1.3.6.1.2.1.2.2.1.6.2
iso.3.6.1.2.1.2.2.1.6.2 = Hex-STRING: FA 16 3E 99 46 D1

--- a/plugins/inputs/snmp/snmp.go
+++ b/plugins/inputs/snmp/snmp.go
@@ -716,7 +716,6 @@ func (s *Snmp) getConnection(agent string) (snmpConnection, error) {
 //  "hwaddr" will convert the value into a MAC address.
 //  "ipaddr" will convert the value into into an IP address.
 //  "" will convert a byte slice into a string.
-// Any other conv will return the input value unchanged.
 func fieldConvert(conv string, v interface{}) (interface{}, error) {
 	if conv == "" {
 		if bs, ok := v.([]byte); ok {
@@ -805,6 +804,7 @@ func fieldConvert(conv string, v interface{}) (interface{}, error) {
 		default:
 			return nil, fmt.Errorf("invalid type (%T) for hwaddr conversion", v)
 		}
+		return v, nil
 	}
 
 	if conv == "ipaddr" {
@@ -829,7 +829,7 @@ func fieldConvert(conv string, v interface{}) (interface{}, error) {
 		return v, nil
 	}
 
-	return v, nil
+	return nil, fmt.Errorf("invalid conversion type '%s'", conv)
 }
 
 // snmpTranslate resolves the given OID.

